src/SqlFormatter.php

```diff
@@ -722,343 +43,18 @@ final class SqlFormatter
-        // 4. single quoted string using '' or \' to escape
-        if (preg_match(
-            '/^(((`[^`]*($|`))+)|
-            ((\[[^\]]*($|\]))(\][^\]]*($|\]))*)|
-            (("[^"\\\\]*(?:\\\\.[^"\\\\]*)*("|$))+)|
-            ((\'[^\'\\\\]*(?:\\\\.[^\'\\\\]*)*(\'|$))+))/sx',
-            $string,
-            $matches
-        )) {
-            $ret = $matches[1];
-        }
-
-        return $ret;
-    }
-
-    /**
-     * Takes a SQL string and breaks it into tokens.
-     * Each token is an associative array with type and value.
-     *
-     * @param string $string The SQL string
-     *
-     * @return mixed[][] An array of tokens.
-     */
-    private function tokenize(string $string) : array
-    {
-        $tokens = [];
-
-        // Used for debugging if there is an error while tokenizing the string
-        $originalLength = strlen($string);
-
-        // Used to make sure the string keeps shrinking on each iteration
-        $oldStringLen = strlen($string) + 1;
-
-        $token = null;
-
-        $currentLength = strlen($string);
-
-        // Keep processing the string until it is empty
-        while ($currentLength) {
-            // If the string stopped shrinking, there was a problem
-            if ($oldStringLen <= $currentLength) {
-                $tokens[] = [
-                    self::TOKEN_VALUE=>$string,
-                    self::TOKEN_TYPE=>self::TOKEN_TYPE_ERROR,
-                ];
-
-                return $tokens;
-            }
-
-            $oldStringLen =  $currentLength;
-
-            // Determine if we can use caching
-            if ($currentLength >= $this->maxCachekeySize) {
-                $cacheKey = substr($string, 0, $this->maxCachekeySize);
-            } else {
-                $cacheKey = false;
-            }
-
-            // See if the token is already cached
-            if ($cacheKey && isset($this->tokenCache[$cacheKey])) {
-                // Retrieve from cache
-                $token       = $this->tokenCache[$cacheKey];
-                $tokenLength = strlen($token[self::TOKEN_VALUE]);
-                $this->cacheHits++;
-            } else {
-                // Get the next token and the token type
-                $token       = $this->getNextToken($string, $token);
-                $tokenLength = strlen($token[self::TOKEN_VALUE]);
-                $this->cacheMisses++;
-
-                // If the token is shorter than the max length, store it in cache
-                if ($cacheKey && $tokenLength < $this->maxCachekeySize) {
-                    $this->tokenCache[$cacheKey] = $token;
-                }
-            }
-
-            $tokens[] = $token;
-
-            // Advance the string
-            $string = substr($string, $tokenLength);
-
-            $currentLength -= $tokenLength;
-        }
-
-        return $tokens;
+    public function __construct(
+        ?Tokenizer $tokenizer = null,
+        ?Highlighter $highlighter = null
```

Comment:
Reminder: i still think that this param should go away and be passes in the specific method when possible

Reply:
I'm not forgetting, I have #25 as a reminder ;)